destination.go

```diff
@@ -1,77 +1,203 @@
+	"errors"
 	"fmt"
 
 	"github.com/conduitio/conduit-commons/config"
 	"github.com/conduitio/conduit-commons/opencdc"
 	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/sashabaranov/go-openai"
 )
 
 type Destination struct {
 	sdk.UnimplementedDestination
 
 	config DestinationConfig
+	client *openai.Client
 }
 
+//go:generate paramgen -output=paramgen_dest.go DestinationConfig
+
 type DestinationConfig struct {
-	// Config includes parameters that are the same in the source and destination.
-	Config
-	// DestinationConfigParam must be either yes or no (defaults to yes).
-	DestinationConfigParam string `validate:"inclusion=yes|no" default:"yes"`
+	// APIKey is the OpenAI api key to use for the api client.
+	APIKey string `json:"api_key" validate:"required"`
+
+	// VectorStoreID is the id of the vector store to write records into.
+	VectorStoreID string `json:"vector_store_id" validate:"required"`
 }
 
 func NewDestination() sdk.Destination {
-	// Create Destination and wrap it in the default middleware.
-	return sdk.DestinationWithMiddleware(&Destination{}, sdk.DefaultDestinationMiddleware()...)
+	disable := false
+	return sdk.DestinationWithMiddleware(&Destination{}, sdk.DefaultDestinationMiddleware(sdk.DestinationWithSchemaExtractionConfig{
+		PayloadEnabled: &disable,
+		KeyEnabled:     &disable,
+	})...)
 }
 
 func (d *Destination) Parameters() config.Parameters {
-	// Parameters is a map of named Parameters that describe how to configure
-	// the Destination. Parameters can be generated from DestinationConfig with
-	// paramgen.
 	return d.config.Parameters()
 }
 
 func (d *Destination) Configure(ctx context.Context, cfg config.Config) error {
-	// Configure is the first function to be called in a connector. It provides
-	// the connector with the configuration that can be validated and stored.
-	// In case the configuration is not valid it should return an error.
-	// Testing if your connector can reach the configured data source should be
-	// done in Open, not in Configure.
-	// The SDK will validate the configuration and populate default values
-	// before calling Configure. If you need to do more complex validations you
-	// can do them manually here.
-
-	sdk.Logger(ctx).Info().Msg("Configuring Destination...")
+	sdk.Logger(ctx).Info().Msg("configuring destination...")
 	err := sdk.Util.ParseConfig(ctx, cfg, &d.config, NewDestination().Parameters())
 	if err != nil {
 		return fmt.Errorf("invalid config: %w", err)
 	}
 	return nil
 }
 
-func (d *Destination) Open(_ context.Context) error {
-	// Open is called after Configure to signal the plugin it can prepare to
-	// start writing records. If needed, the plugin should open connections in
-	// this function.
+func (d *Destination) Open(ctx context.Context) error {
+	d.client = openai.NewClient(d.config.APIKey)
+
+	// check that the passed api key is valid, so that we ensure that the api
+	// calls have no auth errors.
+
+	if _, err := d.client.ListModels(ctx); err != nil {
```

Comment:
Out of curiosity, do we need to make a specific client request to validate the API key? I wonder if we could simply error on the next `RetrieveVectorStore` client request.

Reply:
Agree, we can validate both parameters at once actually.